modules/virt-attaching-vm-to-ovn-secondary-nw-cli.adoc

```diff
@@ -0,0 +1,79 @@
+// Module included in the following assemblies:
+//
+// * virt/vm_networking/virt-connecting-vm-to-ovn-secondary-network.adoc              
+
+:_content-type: PROCEDURE                                             
+[id="virt-attaching-vm-to-ovn-secondary-nw-cli_{context}"]                                   
+= Attaching a virtual machine to an OVN-Kubernetes secondary network using the CLI
+
+You can connect a virtual machine (VM) to the OVN-Kubernetes secondary network by including the network details in the VM configuration.
+
+.Prerequisites
+* You have access to the cluster as a user with `cluster-admin` privileges.
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+. Edit the `VirtualMachine` manifest to add the OVN-Kubernetes secondary network interface details, as in the following example:
++
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: vm-server
+spec:
+  running: true
+  template:
+    spec:
+      domain:
+        devices:
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio
+            - name: cloudinitdisk
+              disk:
+                bus: virtio
+          interfaces:
+          - name: default
+            masquerade: {}
+          - name: flatl2-overlay
+            bridge: {}
+        machine:
+          type: ""
+        resources:
+          requests:
+            memory: 1024M
+      networks:
+      - name: default
+        pod: {}
+      - name: flatl2-overlay
+        multus:
+          networkName: l2-network <1>
+      terminationGracePeriodSeconds: 0
+      volumes:
+        - name: containerdisk
+          containerDisk:
+            image: quay.io/kubevirt/fedora-with-test-tooling-container-disk:devel
```

Comment:
maybe we should use a newer image here ? This one is ooooooold.

Maybe use the same one we're using elsewhere ?

Reply:
We actually do not include the `volumes` stanza in the SR-IOV or Linux bridge documentation. What are the most relevant sections of the `VirtualMachine` manifest that we need to include in this example?

Reply:
I had assumed this would be a copy / paste able example. Looking at the other examples - [bridge](https://docs.openshift.com/container-platform/4.14/virt/vm_networking/virt-connecting-vm-to-linux-bridge.html#virt-attaching-vm-secondary-network-cli_virt-connecting-vm-to-linux-bridge) / [sriov](https://docs.openshift.com/container-platform/4.14/virt/vm_networking/virt-connecting-vm-to-sriov.html#virt-attaching-vm-to-sriov-network_virt-connecting-vm-to-sriov) - it looks it isn't.

I would argue the `cloudinitdisk` is a must since that is how the user would configure the IP on the interface - which would be borderline useless without it... 

But the same can be said about the other examples (none are showing cloud init configuration), and the name of the article / heading is `Connecting a VM to an XYZ network` . It says nothing about L3. Thus, I would keep it simple and do something like:

```yaml
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachine
metadata:
  name: vm-server
spec:
  template:
    spec:
      domain:
        devices:
          interfaces:
          - name: default
            masquerade: {}
          - name: flatl2-overlay (1)
            bridge: {}
      networks:
      - name: default
        pod: {}
      - name: flatl2-overlay  (2)
        multus:
          networkName: l2-network (3)
```

I would add procedures pointing out the name of the interface **must** match the name of the network, as done in [bridge](https://docs.openshift.com/container-platform/4.14/virt/vm_networking/virt-connecting-vm-to-linux-bridge.html#virt-attaching-vm-secondary-network-cli_virt-connecting-vm-to-linux-bridge) / [sriov](https://docs.openshift.com/container-platform/4.14/virt/vm_networking/virt-connecting-vm-to-sriov.html#virt-attaching-vm-to-sriov-network_virt-connecting-vm-to-sriov).

Reply:
Done

Reply:
FWIW, I think you forgot to push the new changes. I still see the "old" code - with the img.